app/models/order.rb

```diff
@@ -14,7 +18,11 @@ def item_count
     items.sum(:quantity)
   end
 
-  def self.by_status
-    order(status: :asc)
+  def cancel
+    update(status: "cancelled")
+    item_orders.where(status: "fulfilled").each do |item_order|
+      item_order.item.increment!(:inventory, item_order.quantity)
+    end
+    item_orders.update(status: "unfulfilled")
```

Comment:
This is a complicated one to solve from a pure AR perspective. Maybe we could go over it together in Postico using the fancy button in the top-left?